lib/pbench/test/unit/agent/test_utils.py

```diff
@@ -0,0 +1,210 @@
+
+        with pytest.raises(OurServer.Err) as excinfo:
+            bs = OurServer("bad_host.example.com", _def)
+        assert excinfo.value.return_code == 43, f"{excinfo.value!r}"
+
+        with pytest.raises(OurServer.Err) as excinfo:
+            bs = OurServer("bad-port.example.com:bad", _def)
+        assert excinfo.value.return_code == 42, f"{excinfo.value!r}"
+
+        for arg1, exp_port in [
+            ("host.example.com:2345", 2345),
+            ("host.example.com:", 4242),
+        ]:
+            bs = OurServer(arg1, _def)
+            assert bs.pid_file is None
+            assert bs.host == "host.example.com"
+            assert bs.port == exp_port
+            assert bs.bind_host == bs.host
+            assert bs.bind_port == bs.port
+            assert repr(bs) == f"forty-two - host.example.com:{exp_port:d}"
+
+        bs = OurServer(":2345", _def)
+        assert bs.pid_file is None
+        assert bs.host == "localhost"
+        assert bs.port == 2345
+        assert bs.bind_host == bs.host
+        assert bs.bind_port == bs.port
+        assert repr(bs) == "forty-two - localhost:2345"
+
+        bs = OurServer("127.0.0.42:4567", _def)
+        assert bs.pid_file is None
+        assert bs.host == "127.0.0.42"
+        assert bs.port == 4567
+        assert bs.bind_host == bs.host
+        assert bs.bind_port == bs.port
+        assert repr(bs) == "forty-two - 127.0.0.42:4567"
+
+        bs = OurServer("[127::42]:4567", _def)
+        assert bs.pid_file is None
+        assert bs.host == "127::42"
+        assert bs.port == 4567
+        assert bs.bind_host == bs.host
+        assert bs.bind_port == bs.port
+        assert repr(bs) == "forty-two - 127::42:4567"
+
+        bs = OurServer("bind.example.com:2345;host.example.com:6789", _def)
+        assert bs.pid_file is None
+        assert bs.host == "host.example.com"
+        assert bs.port == 6789
+        assert bs.bind_host == "bind.example.com"
+        assert bs.bind_port == 2345
+        assert repr(bs) == "forty-two - bind.example.com:2345 / host.example.com:6789"
+
+    def test_kill(self, pytestconfig, monkeypatch):
+        bs = OurServer("localhost", "localhost")
+        with pytest.raises(AssertionError):
+            bs.kill(1)
+
+        bs = OurServer("localhost", "localhost")
+        TMP = pathlib.Path(pytestconfig.cache.get("TMP", None))
+        pidfile = TMP / "test.pid"
+        pidfile.write_text("12345")
+        bs.pid_file = pidfile
+        ret = bs.kill(42)
+        assert ret == 42
+
+        pidfile.write_text("badpid")
+        ret = bs.kill(42)
+        assert ret == (BaseReturnCode.KILL_BADPID * 100) + 42
+
+        bs.pid_file = TMP / "enoent.pid"
+        ret = bs.kill(42)
+        assert ret == 42
+
+        class MockPath:
+            def __init__(self, kind):
+                self.kind = kind
+
+            def read_text(self):
+                if self.kind == "os":
+                    raise OSError("fake oserror")
```

Comment:
That doesn't look much like an [`OSError`](https://docs.python.org/3/library/exceptions.html#OSError) -- it should have an `errno` code as it's first argument, I think.

Reply:
Rather than having the c'tor accept a string as an argument (type hints would be handy, BTW), how about having it accept the exception object to raise?  That would decouple and simplify things a bit.

Reply:
Fixed as a separate commit.